apisix/stream/router/ip_port.lua

```diff
@@ -33,14 +35,38 @@ local _M = {version = 0.1}
 
 local function match_addrs(route, vars)
     -- todo: use resty-ipmatcher to support multiple ip address
-    if route.value.remote_addr and
-       route.value.remote_addr ~= vars.remote_addr then
-        return false
+    if route.value.remote_addr then
+        local ip = remote_addr_match_cache[route.value.remote_addr]
```

Comment:
1. It's dangerous to use a module-level table to cache the match result, since it may cause the memory leak;
2. The IP match overheads are not so high, the caching is unnecessary;

Reply:
Make sense, I'll update mycode